### PR TITLE
Add vehicle chargings and trips API endpoint

### PR DESCRIPTION
**Pull request by GewoonJaap**
This PR adds endpoints to retrieve vehicles charging sessions and the trips.

### CODE REVIEW

**Comment by GewoonJaap** [comment]
Also added CURL to the Docker image to support HTTP healthchecks for the container.

**Comment by flobz** [comment]
thanks for you contribution I will look at it.

**Comment by GewoonJaap** [comment]
Thank you @flobz I see the pipeline fails now because I added the curl package to enable docker HTTP health checks. Should I pin curl to a version?

**Comment by vineethktpla** [comment]
Very useful and good work @GewoonJaap . Was waiting for this for a long time and I hope it will merge soon.
This will ease my work in home assistant to track the trips. I am currently using trigger based automations to calculate and my trips and charging sessions

File: psa_car_controller/web/view/api.py [hunk 1]
```diff
@@ -181,6 +186,27 @@ def settings_section(section: str):
         APP.config.write_config()
     return json_response(config_section.json())
 
+@app.route('/get_vehicletrips')
```
Comment (flobz) [inline psa_car_controller/web/view/api.py:189]:
```suggestion
@app.route('/vehicles/trips')
```
wdyt ?

File: psa_car_controller/web/view/api.py [hunk 2]
```diff
@@ -181,6 +186,27 @@ def settings_section(section: str):
...
+        trips_by_vin = Trips.get_trips(Cars([car]))
+        trips = trips_by_vin[car.vin]
+        trips_as_dict = trips.get_trips_as_dict()
+        return FlaskResponse(json.dumps(trips_as_dict, default=str), mimetype='application/json')
+    except (IndexError, TypeError):
+        logger.debug("Failed to get trips, there is probably not enough data yet:", exc_info=True)
+        return FlaskResponse(json.dumps([]), mimetype='application/json')
+
+
+@app.route('/get_vehiclechargings')
```
Comment (flobz) [inline psa_car_controller/web/view/api.py:202]:
```suggestion
@app.route('/vehicles/chargings')
```

File: psa_car_controller/web/view/api.py [hunk 3]
```diff
@@ -181,6 +186,27 @@ def settings_section(section: str):
...
     return json_response(config_section.json())
 
+@app.route('/get_vehicletrips')
+def get_trips():
+    try:
+        car = APP.myp.vehicles_list[0]
+        trips_by_vin = Trips.get_trips(Cars([car]))
+        trips = trips_by_vin[car.vin]
+        trips_as_dict = trips.get_trips_as_dict()
+        return FlaskResponse(json.dumps(trips_as_dict, default=str), mimetype='application/json')
```
Comment (flobz) [inline psa_car_controller/web/view/api.py:196]: Could you use jsonify like this :
```suggestion
        return jsonify(trips_as_dict)
```

**Review by flobz — CHANGES_REQUESTED** [review]
Thanks for you contribution.
Could you look to my suggested changes ?

File: psa_car_controller/web/view/api.py (hunk 3 above)
Comment (GewoonJaap) [reply psa_car_controller/web/view/api.py]: Yeah that looks better indeed! I just updated :)

**Comment by GewoonJaap** [comment]
> Thanks for you contribution. Could you look to my suggested changes ?

Updated the API and readme with the new endpoints and changed the JSON return method as you suggested :)

**Comment by flobz** [comment]
lgtm thanks !

**Pull request merged**